app/models/manageiq/providers/ansible_tower/automation_manager/machine_credential/api_create.rb

```diff
@@ -0,0 +1,36 @@
+class ManageIQ::Providers::AnsibleTower::AutomationManager::MachineCredential
+  module ApiCreate
+    def create_in_provider(manager_id, params)
+      manager = ExtManagementSystem.find(manager_id)
+      credential = manager.with_provider_connection do |connection|
+        connection.api.credentials.create!(params)
```

Comment:
I don't like this at all and that's why I have only done this to `MachineCredential`. I prefer this to be done at a higher level class.  The existing `AutomationManager::Authentication` is not the right place because this particular behavior is Tower specific.  
We need to insert another `Authentication` in the name space of Tower. Are you onboard with that?